docs/writing-policies/spec/05-context-aware-policies.mdx

```diff
@@ -0,0 +1,341 @@
+  ```
+  
+  The `req` parameter contains the filter arguments used to find the requested resources. It is reference to a `ListResourcesByNamespaceRequest` which the fields are:
+  
+  ```go
+  // Set of parameters used by the `list_resources_by_namespace` function
+  type ListResourcesByNamespaceRequest struct {
+  	// apiVersion of the resource (v1 for core group, groupName/groupVersions for other).
+  	APIVersion string `json:"api_version"`
+  	// Singular PascalCase name of the resource
+  	Kind string `json:"kind"`
+  	// Namespace scoping the search
+  	Namespace string `json:"namespace"`
+  	// A selector to restrict the list of returned objects by their labels.
+  	// Defaults to everything if omitted
+  	LabelSelector string `json:"label_selector,omitempty"`
+  	// A selector to restrict the list of returned objects by their fields.
+  	// Defaults to everything if omitted
+  	FieldSelector string `json:"field_selector,omitempty"`
+  }
+  ```
+  
+  ### GetResource
+  
+  Function to request information about a given resource. The function definition is:
+  
+  ```go
+  // GetResource gets a specific Kubernetes resource.
+  func (h *Host) GetResource(req GetResourceRequest) ([]byte, error) { ... }
+  ```
+  
+  The `req` parameter contains the filter arguments used to find the requested resource. It is reference to a `GetResourceRequest` which the fields are:
+  
+  ```go
+  // Set of parameters used by the `get_resource` function
+  type GetResourceRequest struct {
+  	APIVersion string `json:"api_version"`
+  	// Singular PascalCase name of the resource
+  	Kind string `json:"kind"`
+  	// Namespace scoping the search
+  	Namespace string `json:"namespace"`
+  	// The name of the resource
+  	Name string `json:"name"`
+  	// Disable caching of results obtained from Kubernetes API Server
+  	// By default query results are cached for 5 seconds, that might cause
+  	// stale data to be returned.
+  	// However, making too many requests against the Kubernetes API Server
+  	// might cause issues to the cluster
+  	DisableCache bool `json:"disable_cache"`
+  }
+  ```
+
+  </TabItem>
+</Tabs>
+
+
+The workflow we have seen until now has been that the policy exposes
+well known waPC functions, namely: `validate` and
+`validate_settings`. At some point, the host will call these functions
+when it requires either to validate a request, or to validate the
+settings that were provided to it for the given policy.
+
+In this case, after the host calls to the `validate` waPC function in
+the guest, the guest is able to retrieve cluster information in order
+to produce a response to the `validate` waPC function that was called
+by the host on the guest.
+
+This guest-host intercommunication is performed using the regular waPC
+host calling mechanism, and so any guest implementing the waPC
+intercommunication mechanism is able to request this information from
+the host.
+
+waPC has the following function arguments when performing a call from
```

Comment:
I feel this portion of the document is a repetition of what you created under the `host-capabilities` section. I guess this comes from the "old" version of our docs. I would drop it entirely